src/components/EventExpenses.tsx

```diff
@@ -0,0 +1,172 @@
+    client: client,
+    fetchPolicy: "network-only",
+    variables: {
+      eventID,
+    },
+  });
+
+  const { data: expensesByCategory } = useQuery(GET_EXPENSES_BY_CATEGORY, {
+    client: client,
+    fetchPolicy: "network-only",
+    variables: {
+      eventID,
+    },
+  });
+  const categoriesWiseExpenses = expensesByCategory?.getExpensesByCategory;
+
+  const [deleteExpense] = useMutation(DELETE_EXPENSE, {
+    client: client,
+    refetchQueries: [GET_EXPENSES_OF_EVENT, "GetExpensesOfEvent"],
+  });
+  const { toast } = useToast();
+
+  const deleteExpenseHandler = async (id: string) => {
+    try {
+      const res = await deleteExpense({
+        variables: {
+          id,
+        },
+      });
+      if (res.data) {
+        toast({
+          title: "Success",
+          description: "Expense deleted successfully.",
+          variant: "success",
+        });
+      }
+    } catch (err: any) {
+      toast({
+        title: "Error",
+        description: err?.message,
+        variant: "destructive",
+      });
+    }
+  };
+
+  const editHandler = (data: any) => {
+    router.push({
+      pathname: `/dashboard/expense/add`,
+      query: {
+        event: eventID,
+        disable: true,
+        mode: "edit",
+        ...data,
+      },
+    });
+  };
+
+  let data = [];
+  if (expenses?.getExpensesOfEvent?.length > 0) {
+    data = expenses?.getExpensesOfEvent?.map((el: any) => {
+      return {
+        ...el,
+        action: (
+          <>
+            <button
+              onClick={() => editHandler(el)}
+              className="bg-indigo-600 p-1 text-sm text-white px-2 py-1 m-2"
+            >
+              <BiEdit size={18} />
+            </button>
+            <AlertDialog>
+              <AlertDialogTrigger>
+                <button className="bg-red-600 p-1 text-sm text-white px-2 py-1 my-2">
+                  <AiOutlineDelete size={18} />
+                </button>
+              </AlertDialogTrigger>
+              <AlertDialogContent>
+                <AlertDialogHeader>
+                  <AlertDialogTitle>Are you absolutely sure?</AlertDialogTitle>
+                  <AlertDialogDescription>
+                    This action cannot be undone. This will permanently delete
+                    your account and remove your data from our servers.
+                  </AlertDialogDescription>
+                </AlertDialogHeader>
+                <AlertDialogFooter>
+                  <AlertDialogCancel>Cancel</AlertDialogCancel>
+                  <AlertDialogAction
+                    className="bg-red-600 hover:bg-red-900"
+                    onClick={() => {
+                      deleteExpenseHandler(el.id);
+                    }}
+                  >
+                    Continue
+                  </AlertDialogAction>
+                </AlertDialogFooter>
+              </AlertDialogContent>
+            </AlertDialog>
+          </>
+        ),
+      };
+    });
+  }
```

Comment:
Better to extract out this to a different component `ExpenseItem` inside this `EventExpense` component or outside, rather than assigning this to a variable to make it clear.